File: Java/프로그래머스/0/120909. 제곱수 판별하기/제곱수 판별하기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        for(int i=2; i<=n; i++){
            if(i*i!=n) answer=2;
            if(i*i==n) {
            answer=1;
            break;
            }
        }
        return answer;
    }
}

File: C++/프로그래머스/0/120909. 제곱수 판별하기/제곱수 판별하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 2;
    int i=1;
    while(true){
        if(i>=n) break;
        if(n==i*i){
            answer=1;
            break;
        }
        i++;
    }
    return answer;
}